src/cli.rs

```diff
@@ -0,0 +1,19 @@
+use clap::Parser;
+use std::path::PathBuf;
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+pub struct CliArgs {
+    #[clap(short, long)]
+    pub config_path: Option<PathBuf>,
+}
+
+pub enum CliCommand {
+    ConfigPath(PathBuf),
+}
+///This function checks for any commands passed to the program.
+///Return the arguments passed to the program if any.
+pub fn check_args() -> Option<CliArgs> {
+    let args = CliArgs::parse();
+    Some(args)
+}
```

Comment:
This whole approach is wrong.
Can we do a `Default::default` thing for the `config_path` to be the original config path? Just get the `.parent()` or something like this.

In the [clap docs](https://docs.rs/clap/latest/clap/), we have

```rs

    /// Number of times to greet
    #[arg(short, long, default_value_t = 1)]
    count: u8,
}
```

Maybe we can do something similar.